### PR TITLE
refactor: Socket optimization and standardisation. 

### PR DESCRIPTION
**Pull request by debendraoli**
This pr standardises the socket response format and couple of refactor the how block search is done.

### CODE REVIEW

File: relayer/provider/provider.go [hunk 1]
```diff
@@ -39,7 +40,7 @@ type ChainProvider interface {
 	SetAdmin(context.Context, string) error
 
 	FinalityBlock(ctx context.Context) uint64
-	GenerateMessages(ctx context.Context, messageKey *types.MessageKeyWithMessageHeight) ([]*types.Message, error)
+	GenerateMessages(ctx context.Context, fromHeight, toHeight uint64) ([]*types.Message, error)
```
Comment (bcsainju) [inline relayer/provider/provider.go:43]: Should we inform or document somewhere before chaing the behavior of existing command and replacing with the new one ?? any reasons for this change as we move from specific relay message to range which has its own downsides.

File: relayer/socket/server.go [hunk 2]
```diff
@@ -89,292 +88,256 @@ func (s *Server) send(conn net.Conn, data []byte) error {
...
-			}
-			data, err := jsoniter.Marshal(&ResRelayMessage{types.NewRouteMessage(msgs[0])})
-			if err != nil {
-				return nil, err
-			}
-			return &Message{EventRelayMessage, data}, nil
+		if req.FromHeight > req.ToHeight {
+			return response.SetError(fmt.Errorf("fromHeight should be less than toHeight"))
+		} else if req.ToHeight-req.FromHeight > 100 {
+			return response.SetError(fmt.Errorf("fromHeight and toHeight should be less than 100"))
```
Comment (bcsainju) [inline relayer/socket/server.go:162]: the error message is misleading please correct.

File: relayer/relay.go [hunk 3]
```diff
@@ -12,6 +12,7 @@ import (
 )
 
 var (
+	Version                   = "dev"
```
[the remark this answers is not in the document]
Comment (debendraoli) [reply relayer/relay.go:15]: yes.

File: relayer/socket/types.go [hunk 4]
```diff
@@ -62,8 +76,8 @@ type ResMessageRemove struct {
 }
 
 type ResMessageList struct {
-	Messages []*types.RouteMessage `json:"messages"`
-	Total    int                   `json:"total"`
+	Message []*types.RouteMessage `json:"message"`
```
[the remark this answers is not in the document]
Comment (debendraoli) [reply relayer/socket/types.go:79]: fields and struct should not be plural named, following idiomatic go.

File: relayer/provider/provider.go (hunk 1 above)
Comment (debendraoli) [reply relayer/provider/provider.go:43]: It does not change the inner working. messageKey struct input is somewhat redundant.

File: relayer/socket/server.go (hunk 2 above)
Comment (sherpalden) [reply relayer/socket/server.go]: fixed

File: relayer/socket/server.go [hunk 5]
```diff
@@ -58,23 +59,20 @@ func (s *Server) server(c net.Conn) {
...
 }
 
 // makeError for the client to write to socket
 func makeError(err error) []byte {
-	message := &Message{EventError, []byte(fmt.Sprintf(`{"message": "%s"}`, err.Error()))}
+	message := &Response{Event: EventError, Message: err.Error()}
 	data, err := jsoniter.Marshal(message)
 	if err != nil {
-		return []byte(fmt.Sprintf(`{"error": "%s"}`, err.Error()))
+		return []byte(fmt.Sprintf(`{"event":"%s","message":"%s","success":false}`, EventError, err.Error()))
```
[the remark this answers is not in the document]
Comment (sherpalden) [reply relayer/socket/server.go:76]: This is fallback error Response when failed to serialize actual response.

**Review by sherpalden — APPROVED** [review]

**Pull request merged**